minpy/nn/solver.py

```diff
@@ -178,14 +193,27 @@ def loss_func(*params):
 
         self.loss_history.append(loss.asnumpy())
 
+        if self.visualize:
+            grad_norm = 0
+
         # Perform a parameter update
         for p, w in self.model.params.items():
             dw = grads[p]
+            if self.visualize:
+                norm = dw ** 2
+                while not isinstance(norm, minpy.array.Number):
+                    norm = sum(norm)
```

Comment:
I would like to separate the visualize logic here to a separate `GradientVisualizer` object or class. It is an object or an string that is passed to the solver during initialization (rather than a bool flag). It is None by default, but could have `"l2norm"` as one option. Therefore, you could have `np.sum(dw ** 2)` implemented in another file under the same directory as `solver.py`. This makes the module much cleaner and support customized visualizer in the future.